### ISSUE TITLE
Switching rapidly between branches creates unwanted changes

### ISSUE THREAD

**Issue by xuv** [kactus-io/kactus#99]
## Description
When switching rapidly between two branches, the state of first branch is going to over write the state of the second branch and create a pending change to commit.

## Version

* Kactus: 0.3.6

* Operating system: Mac 0SX 10.13.4

## Steps to Reproduce

1. Create repo and add a sketch file.
2. Draw a rectangle in the sketch file and commit.
3. Create new branch called "color".
4. Change color of rectangle and commit.
5. In kactus, switch back and forth between the two branches quickly. After one or two back and forth, changes of one branch will have been applied to the other.

### Expected Behavior

Branches should not "pollute" each other. The changes in one branch should never be applied the other unless requested.

### Actual Behavior

The state of the file in one branch will overwrite the state of the Sketch file in the other branch.

### Logs

[2018-05-15.kactus.production.log](https://github.com/kactus-io/kactus/files/2005513/2018-05-15.kactus.production.log)

**Comment by iphipps** [comment]
I am able to reproduce this consistently without switching quickly.  Switching slowly.

**Version**

Operating system: Mac OSX 10.13.5
Kactus: 0.3.10

**Additional Information**

As soon as I check out a branch, the diff shows:
```
diff --git a/experiment-kactus.sketch b/experiment-kactus.sketch
index 66826a8..04a6305 100644
Binary files a/experiment-kactus.sketch and b/experiment-kactus.sketch differ
```

To check out another branch I must reset head from the command line.

**Comment by baakoo** [comment]
Hi guys, die you tell git, that git should ignore the sketch files? Create/Edit file .gitignore and add *.sketch to it
See here at the bottom of the page ... https://kactus.io/help/how-to/

**Comment by mathieudutour** [comment]
I think this should be fixed in 0.3.17. Feel free to re-open if that's not the case
